cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -639,6 +646,15 @@ private void sendMetricsDelete(String url) {
     }
   }
 
+  private Iterable<ProgramSpecification> getProgramSpecs(Id.Application appId) throws OperationException {
```

Comment:
This was carved out of deleteProgramLocations which is in AppLifecycleHttpHandler.